user_connections.go

```diff
@@ -0,0 +1,239 @@
+		defer m.lock.Unlock()
+		delete(m.handlers, conn.Nick)
+	}
+}
+
+// UserHandler manages connections of a single connected user
+type UserHandler struct {
+	Nick                  string
+	connections           map[*SockChatWS]bool
+	createChannelRequests chan *CreateChannel
+	joinChannelRequests   chan *JoinChannel
+	leaveChannelRequests  chan *LeaveChannel
+	sendMessageRequests   chan *SendMessage
+	lock                  sync.RWMutex
+}
+
+type CreateChannel struct {
+	store       SockchatChannelStore
+	name        string
+	errCallback chan error
+}
+
+type JoinChannel struct {
+	store       SockchatChannelStore
+	channel     string
+	errCallback chan error
+}
+
+type LeaveChannel struct {
+	store       SockchatChannelStore
+	channel     string
+	errCallback chan error
+}
+
+type SendMessage struct {
+	store       SockchatChannelStore
+	channel     string
+	message     string
+	errCallback chan error
+}
+
+func NewUserHandler(conn *SockChatWS) *UserHandler {
+	handler := UserHandler{
+		Nick:                  conn.Nick,
+		connections:           make(map[*SockChatWS]bool),
+		createChannelRequests: make(chan *CreateChannel),
+		joinChannelRequests:   make(chan *JoinChannel),
+		leaveChannelRequests:  make(chan *LeaveChannel),
+		sendMessageRequests:   make(chan *SendMessage),
+	}
+	go handler.HandleRequests()
+	return &handler
+}
+
+func (u *UserHandler) HandleRequests() {
+	for {
+		select {
+		case req := <-u.createChannelRequests:
+			err := req.store.CreateChannel(req.name)
+			req.errCallback <- err
+
+		case req := <-u.joinChannelRequests:
+			err := req.store.AddUserToChannel(req.channel, u)
+			if err != nil {
+				req.errCallback <- err
+				continue
+			}
+			req.errCallback <- nil
+
+		case req := <-u.leaveChannelRequests:
+			err := req.store.RemoveUserFromChannel(req.channel, u)
+			if err != nil {
+				req.errCallback <- err
+				continue
+			}
+			req.errCallback <- nil
+
+		case req := <-u.sendMessageRequests:
+			channel, err := req.store.GetChannel(req.channel)
+			if err != nil {
+				req.errCallback <- err
+				continue
+			}
+			if !channel.GetMembers()[u] {
+				req.errCallback <- ErrUserNotInChannel
+				continue
+			}
+			channel.MessageMembers(NewSocketMessage("new_message", MessageEvent{req.message, req.channel, u.Nick}))
+			req.errCallback <- nil
+		}
+	}
+}
+
+func (u *UserHandler) RequestNewChannel(c ChannelRequest, store SockchatChannelStore) error {
+	errCallback := make(chan error)
+	u.createChannelRequests <- &CreateChannel{
+		store:       store,
+		name:        c.Name,
+		errCallback: errCallback,
+	}
+
+	if err := <-errCallback; err != nil {
+		return err
+	}
+	return nil
+}
+
+func (u *UserHandler) RequestJoinChannel(c ChannelRequest, store SockchatChannelStore) error {
+	errCallback := make(chan error)
+	u.joinChannelRequests <- &JoinChannel{
+		store:       store,
+		channel:     c.Name,
+		errCallback: errCallback,
+	}
+	if err := <-errCallback; err != nil {
+		return err
+	}
+	return nil
```

Comment:
```suggestion
	return <-errCallback
```
It can be simplified